smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -106,25 +103,26 @@ public void partitionsAssigned(Context context, Set<TopicPartition> partitions)
 
         offsetStores.clear();
 
-        resetNextCommitTime();
+        if (timerId != null) {
+            context.owner().cancelTimer(timerId);
+            timerId = null;
+        }
+
+        if (!partitions.isEmpty()) {
+            timerId = context
+                    .owner()
+                    .setPeriodic(autoCommitInterval, this::flushAndCheckHealth);
```

Comment:
instead of using a periodic task, we may want to do a timed task starting a new timed task. So the gap between the end of the task and the beginning of the next one will be the configured time. Without it's the time between 2 beginning which might be too close.

Reply:
I wavered between the two options. I'll go with your suggestions.